io/plugins/eu.esdihumboldt.hale.io.codelist.skos/src/eu/esdihumboldt/hale/io/codelist/skos/reader/SkosCodeList.java

```diff
@@ -0,0 +1,449 @@
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+import org.xml.sax.EntityResolver;
+import org.xml.sax.InputSource;
+import org.xml.sax.SAXException;
+
+import de.fhg.igd.slf4jplus.ALogger;
+import de.fhg.igd.slf4jplus.ALoggerFactory;
+import eu.esdihumboldt.hale.common.codelist.CodeList;
+
+/**
+ * Reads a SKOS code list and treat concepts as code entries
+ * 
+ * @author Arun
+ */
+public class SkosCodeList implements CodeList {
+
+	private static final ALogger log = ALoggerFactory.getLogger(SkosCodeList.class);
+
+	private static final String SKOS_URI = "http://www.w3.org/2004/02/skos/core#";
+	private static final String SKOS_PREF_LABEL = SKOS_URI + "prefLabel";
+	private static final String SKOS_DEF_LABEL = SKOS_URI + "definition";
+	private static final String SKOS_TOPCONCEPT_LABEL = SKOS_URI + "topConceptOf";
+	private static final String SKOS_INSCHEME_LABEL = SKOS_URI + "inScheme";
+
+	private static final String USAGENOTE_LABEL = "usageNote";
+
+	private String identifier;
+
+	private String namespace;
+
+	private String description;
+
+	private final URI location;
+
+	private final Map<String, CodeEntry> entriesByName = new LinkedHashMap<String, CodeEntry>();
+
+	private final Map<String, CodeEntry> entriesByIdentifier = new LinkedHashMap<String, CodeEntry>();
+
+	private SKOSDataset dataSet;
+
+	/**
+	 * Create a code list from a RDF file and URL.
+	 * 
+	 * @param in input stream of source
+	 * @param location the location from where code list loaded
+	 * @throws Exception if something will go wrong
+	 */
+	public SkosCodeList(InputStream in, URI location) throws Exception {
+		this.location = location;
+		this.identifier = null;
+		try {
+
+			SKOSManager manager = new SKOSManager();
+			dataSet = manager.loadDatasetFromPhysicalURI(location);
+
+			// get ConceptSchemes
+			if (!getConceptSchemes())
+				if (!getConcepts()) {
+					if (!getConceptsAsXML(in)) {
+						throw new RuntimeException("no concept found!");
+					}
+				}
+
+			this.dataSet = null;
+
+		} catch (Exception ex) {
+			log.error("Error reading skos code list", ex);
+			throw ex;
+		}
+	}
+
+	@Override
+	public Collection<CodeEntry> getEntries() {
+		return new ArrayList<CodeEntry>(entriesByIdentifier.values());
+	}
+
+	@Override
+	public String getNamespace() {
+		return namespace;
+	}
+
+	@Override
+	public String getIdentifier() {
+		if (identifier != null)
+			return identifier;
+
+		if (location != null)
+			return location.toString();
+
+		return null;
+	}
+
+	@Override
+	public String getDescription() {
+		return description;
+	}
+
+	@Override
+	public CodeEntry getEntryByName(String name) {
+		return entriesByName.get(name);
+	}
+
+	@Override
+	public CodeEntry getEntryByIdentifier(String identifier) {
+		return entriesByIdentifier.get(identifier);
+	}
+
+	@Override
+	public URI getLocation() {
+		return location;
+	}
+
+	private boolean getConceptSchemes() {
```

Comment:
All these get(ter) methods w/ boolean or void return type confuse me. Usually this naming convention is used for getters, but these seem to be methods that mostly are there for their side effects.